qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py

```diff
@@ -639,16 +719,31 @@ def _build_qeom_pseudoeigenvalue_problem(
         logger.debug("Build QEOM pseudoeigenvalue problem...")
 
         # 1. Build all EOM operators to evaluate on the ground state
-        tap_eom_matrix_ops = self._build_all_eom_operators(
+        untap_eom_matrix_ops = self._build_all_eom_operators(
             untap_operator,
             expansion_basis_data,
         )
 
+        untap_eom_matrix_ops_sumop = {
+            key: PauliSumOp(op) for key, op in untap_eom_matrix_ops.items()
+        }
+
+        if isinstance(self.qubit_converter, QubitConverter):
+            tap_eom_matrix_ops_sumop = self.qubit_converter.symmetry_reduce_clifford(
+                untap_eom_matrix_ops_sumop
+            )
+        elif isinstance(self.qubit_converter, TaperedQubitMapper):
+            tap_eom_matrix_ops_sumop = self.qubit_converter.taper_clifford(
+                untap_eom_matrix_ops_sumop, suppress_none=True
+            )
+        else:
+            tap_eom_matrix_ops_sumop = untap_eom_matrix_ops_sumop
```

Comment:
Just like before, this same logic also appears multiple times. Extracting this into a separate private method would be good, too